kani-driver/src/metadata.rs

```diff
@@ -158,40 +165,22 @@ pub fn mock_proof_harness(
 /// Search for a proof harness with a particular name.
 /// At the present time, we use `no_mangle` so collisions shouldn't happen,
 /// but this function is written to be robust against that changing in the future.
-fn find_proof_harness<'a>(
-    name: &str,
-    harnesses: &'a [&HarnessMetadata],
-) -> Result<&'a HarnessMetadata> {
-    let mut result: Option<&'a HarnessMetadata> = None;
-    for h in harnesses.iter() {
-        // Either an exact match, or...
-        let matches = h.pretty_name == *name || {
-            // pretty_name will be things like `module::submodule::name_of_function`
-            // and we want people to be able to specify `--harness name_of_function`
-            if let Some(prefix) = h.pretty_name.strip_suffix(name) {
-                prefix.ends_with("::")
-            } else {
-                false
-            }
-        };
-        if matches {
-            if let Some(other) = result {
-                bail!(
-                    "Conflicting proof harnesses named {}:\n {}\n {}",
-                    name,
-                    other.pretty_name,
-                    h.pretty_name
-                );
-            } else {
-                result = Some(h);
-            }
+fn find_proof_harnesses<'a>(
+    targets: BTreeSet<&String>,
+    all_harnesses: &[&'a HarnessMetadata],
+) -> Vec<&'a HarnessMetadata> {
+    debug!(?targets, "find_proof_harness");
+    let mut result = vec![];
+    for md in all_harnesses.iter() {
```

Comment:
What is `md`?

Reply:
metadata... I can expand. The 'h' was annoying me, but I guess `md` is not much better. :)